src/commonMain/kotlin/number/PositiveDouble.kt

```diff
@@ -0,0 +1,222 @@
+     * @param tag An optional [String] used to better localize the failure in case of [ArrowExactPositiveDoubleConstraintError].
+     * @return [PositiveDouble] if the provided [Double] value is positive and finite, otherwise, throws an exception.
+     * @throws [ArrowExactPositiveDoubleConstraintError] when the provided [value] isn't positive and finite.
+     */
+    fun unsafe(
+      value: Double,
+      tag: String = "PositiveDouble#unsafe"
+    ): PositiveDouble = fromDouble(value).fold(
+      ifEmpty = {
+        throw ArrowExactPositiveDoubleConstraintError(
+          violatingValue = value,
+          tag = tag,
+        )
+      },
+      ifSome = { it }
+    )
+
+    /**
+     * Creates a [PositiveDouble] from a [Double] value using the [unsafe] method.
+     *
+     * @param value A [Double] value that must be positive and finite.
+     * @return [PositiveDouble] if the provided [Double] value is positive and finite, otherwise, throws an exception.
+     * @throws [ArrowExactPositiveDoubleConstraintError] when the provided [value] isn't positive and finite.
+     */
+    operator fun invoke(value: Double): PositiveDouble = unsafe(value)
+  }
+
+}
+
+/**
+ * Safely applies the provided function [f] to transform the underlying [PositiveDouble.value].
+ *
+ * The transformation function [f] is used to map the current [PositiveDouble.value] to another [Double].
+ * If the result is both finite and positive, a [Some] containing the new [PositiveDouble] is returned.
+ * If the result is not finite or positive, [None] is returned.
+ *
+ * @param f A transformation function that maps the wrapped [PositiveDouble.value] to another [Double].
+ * @return [Some] containing a new [PositiveDouble] if the result of applying [f] is positive and finite, otherwise [None].
+ *
+ * ### Example:
+ * ```
+ * val positive = PositiveDouble(2.0)
+ * val result = positive.mapSafe { it * 3 } // Some(PositiveDouble(6.0))
+ * val negativeResult = positive.mapSafe { -it } // None
+ * ```
+ */
+fun PositiveDouble.mapSafe(f: (Double) -> Double): Option<PositiveDouble> = fromDouble(f(value))
+
+
+/**
+ * Represents an error that occurs when a value violates the constraint of being both positive and finite.
+ *
+ * This error is thrown when a value that does not meet the requirements of a [PositiveDouble] is passed
+ * to an unsafe API, such as [PositiveDouble.unsafe] or [PositiveDouble.invoke].
+ *
+ * @param violatingValue The [Double] value that violates the positive and finite constraint.
+ * @param tag An optional [String] that provides additional context or information about the source of the error.
+ *
+ * ### Example:
+ * ```
+ * try {
+ *     val invalidPositiveDouble = PositiveDouble(-1.0)
+ * } catch (e: ArrowExactPositiveDoubleConstraintError) {
+ *     println(e.message) // "Value '-1.0' does not meet the constraint 'PositiveDouble', which requires a positive and finite number."
+ * }
+ * ```
+ */
+class ArrowExactPositiveDoubleConstraintError(
```

Comment:
Should we make this less verbose? It's a trade-off between information and verboseness.